pkg/solana/txm/txm.go

```diff
@@ -377,139 +399,209 @@ func (txm *Txm) confirm() {
+		} else {
+			txm.lggr.Debugw("tx failed to move beyond 'processed' within confirm timeout", "id", id, "signature", sig, "timeoutSeconds", txm.cfg.TxConfirmTimeout())
 		}
-		tick = time.After(utils.WithJitter(txm.cfg.ConfirmPollPeriod()))
+	}
+}
+
+// handleConfirmedSignatureStatus handles the case where a transaction signature is in the "confirmed" state on-chain.
+// It updates the transaction state in the local memory.
+func (txm *Txm) handleConfirmedSignatureStatus(sig solanaGo.Signature) {
+	id, err := txm.txs.OnConfirmed(sig)
+	if err != nil && !errors.Is(err, ErrAlreadyInExpectedState) {
+		txm.lggr.Errorw("failed to mark transaction as confirmed", "id", id, "signature", sig, "error", err)
+	} else if err == nil {
+		txm.lggr.Debugw("marking transaction as confirmed", "id", id, "signature", sig)
+	}
+}
+
+// handleFinalizedSignatureStatus handles the case where a transaction signature is in the "finalized" state on-chain.
+// It updates the transaction state in the local memory.
+func (txm *Txm) handleFinalizedSignatureStatus(sig solanaGo.Signature) {
+	id, err := txm.txs.OnFinalized(sig, txm.cfg.TxRetentionTimeout())
+	if err != nil {
+		txm.lggr.Errorw("failed to mark transaction as finalized", "id", id, "signature", sig, "error", err)
+	} else {
+		txm.lggr.Debugw("marking transaction as finalized", "id", id, "signature", sig)
+	}
+}
+
+// rebroadcastExpiredTxs attempts to rebroadcast all transactions that are in broadcasted state and have expired.
+// An expired tx is one where it's blockhash lastValidBlockHeight is smaller than the current slot height.
+// If any error occurs during rebroadcast attempt, they are discarded, and the function continues with the next transaction.
+func (txm *Txm) rebroadcastExpiredTxs(ctx context.Context, client client.ReaderWriter) {
+	currBlockHeight, err := client.GetLatestBlock(ctx)
+	if err != nil || currBlockHeight == nil || currBlockHeight.BlockHeight == nil {
+		txm.lggr.Errorw("failed to get current block height", "error", err)
+		return
+	}
+	// Rebroadcast all expired txes
+	for _, tx := range txm.txs.ListAllExpiredBroadcastedTxs(*currBlockHeight.BlockHeight) {
+		txm.lggr.Debugw("transaction expired, rebroadcasting", "id", tx.id, "signature", tx.signatures, "lastValidBlockHeight", tx.lastValidBlockHeight, "currentBlockHeight", *currBlockHeight.BlockHeight)
+		if len(tx.signatures) == 0 { // prevent panic, shouldn't happen.
+			txm.lggr.Errorw("no signatures found for expired transaction", "id", tx.id)
+			continue
+		}
+		// Removes all signatures associated to tx and cancels context.
+		_, err := txm.txs.Remove(tx.id)
+		if err != nil {
+			txm.lggr.Errorw("failed to remove expired transaction", "id", tx.id, "error", err)
+			continue
+		}
+		rebroadcastTx := pendingTx{
+			tx:  tx.tx,
+			cfg: tx.cfg,
+			id:  tx.id, // using same id in case it was set by caller and we need to maintain it.
+		}
```

Comment:
Just realized since we're just passing in the old tx cfg here, we aren't updating the compute unit price. Think we probably need to update `BaseComputeUnitPrice` in `tx.cfg` before setting it here.
```suggestion
		tx.cfg.BaseComputeUnitPrice = txm.fee.BaseComputeUnitPrice()
		rebroadcastTx := pendingTx{
			tx:  tx.tx,
			cfg: tx.cfg,
			id:  tx.id, // using same id in case it was set by caller and we need to maintain it.
		}
```

Reply:
Oh you are right, thought it was handled by buildTx [here](https://github.com/smartcontractkit/chainlink-solana/blob/0e38174d35b6d6ee9f03dba526bd2bf135ad9809/pkg/solana/txm/txm.go#L257-L264) but it's just using the old value